app/src/main/kotlin/com/wire/android/ui/sharing/ImportMediaViewModel.kt

```diff
@@ -0,0 +1,382 @@
+        is ConversationDetails.Group -> {
+            ConversationItem.GroupConversation(
+                groupName = conversation.name.orEmpty(),
+                conversationId = conversation.id,
+                mutedStatus = conversation.mutedStatus,
+                isLegalHold = legalHoldStatus.showLegalHoldIndicator(),
+                lastMessageContent = lastMessage.toUIPreview(unreadEventCount),
+                badgeEventType = parseConversationEventType(
+                    conversation.mutedStatus,
+                    unreadEventCount
+                ),
+                hasOnGoingCall = hasOngoingCall,
+                isSelfUserCreator = isSelfUserCreator,
+                isSelfUserMember = isSelfUserMember,
+                teamId = conversation.teamId,
+                selfMemberRole = selfRole
+            )
+        }
+
+        is ConversationDetails.OneOne -> {
+            ConversationItem.PrivateConversation(
+                userAvatarData = UserAvatarData(
+                    otherUser.previewPicture?.let { ImageAsset.UserAvatarAsset(wireSessionImageLoader, it) },
+                    otherUser.availabilityStatus,
+                    otherUser.connectionStatus
+                ),
+                conversationInfo = ConversationInfo(
+                    name = otherUser.name.orEmpty(),
+                    membership = userTypeMapper.toMembership(userType),
+                    isSenderUnavailable = otherUser.isUnavailableUser
+                ),
+                conversationId = conversation.id,
+                mutedStatus = conversation.mutedStatus,
+                isLegalHold = legalHoldStatus.showLegalHoldIndicator(),
+                lastMessageContent = lastMessage.toUIPreview(unreadEventCount),
+                badgeEventType = parsePrivateConversationEventType(
+                    otherUser.connectionStatus,
+                    otherUser.deleted,
+                    parseConversationEventType(
+                        conversation.mutedStatus,
+                        unreadEventCount
+                    )
+                ),
+                userId = otherUser.id,
+                blockingState = otherUser.BlockState,
+                teamId = otherUser.teamId
+            )
+        }
+
+        else -> null // We don't care about connection requests
+    }
+
+    private fun searchShareableConversation(conversationDetails: List<ConversationItem>, searchQuery: String): List<ConversationItem> {
+        val matchingConversations =
+            conversationDetails.filter { details ->
+                when (details) {
+                    is ConversationItem.GroupConversation -> details.groupName.contains(searchQuery, true)
+                    is ConversationItem.PrivateConversation -> details.conversationInfo.name.contains(searchQuery, true)
+                    is ConversationItem.ConnectionConversation -> false
+                }
+            }
+        return matchingConversations
+    }
```

Comment:
I suppose this is just for now ? Otherwise I think we should perform an actual search on the SQL layer, what u think ?

Reply:
Hmmm, wouldn't that be way more inneficient? we are observing the conversation details list, so we always have the most updated state of the conversations list. Then we just simply apply here a query/name filter

Reply:
I am not so sure about that, not an expert on SQL :S but if you perform a search on SQL you also have the most up to date data I would say you have it even faster there, but I think we can revisit it once we have the actual implemention of searching for a conversation, because the one we are having now I just a quick solution that is implemented in similar way as this one.

Another point for doing it on SQL level is that we can have a use case that can be consumed by the clients using Kalium, which is pretty important I guess

Reply:
also is this toConversationItem() function called only on ViewModel initilization or each time we search ?

Reply:
This function gets called only once at the beginning, and every time the conversationDetails observes new changes on the list (i.e. conversation order changes) as we are collecting these events as a live flow. 

I don't think that is a major performance issue nor the usability gets affected, but if absolutely needed we could also change it to a single initial api request to get a snapshot of the conversations list at the moment of entering the `ImportMediaScreen`.